source/PlayerEntity.cpp

```diff
@@ -147,6 +171,23 @@ void PlayerEntity::onSimulation(SimTime absoluteTime, SimTime deltaTime) {
 			respawn();
 		}
 	}
+
+	if (!m_gettingMovementInput)
+	{
+		if(movementDeceleration != nullptr)
+		m_acceleratedVelocity = lerp(m_acceleratedVelocity, 0.0f, *movementDeceleration);
+
+		if (m_acceleratedVelocity > 0) {
+			m_linearVector = m_acceleratedVelocity * m_lastDirection;
+		}
+
+		m_headBobCurrentHeight = lerp(m_headBobCurrentHeight, 0.0f, 0.1f);
```

Comment:
Again this fixed lerp will likely be dependent on frame rate (# of times this function is called per second). We should work to make this more uniform by using a time-based metric for interpolation.

Reply:
Updated.